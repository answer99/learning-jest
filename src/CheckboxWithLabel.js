import React, { useState, useCallback } from 'react';

function CheckboxWithLabel({labelOn, labelOff}){

  const [isChecked, setIsChecked] = useState(false);

  const handleOnChange = useCallback(() => {
    setIsChecked(!isChecked);
  }, [isChecked]);

  return (
     <label>
        <input
          value={isChecked}
          type="checkbox"
          checked={isChecked}
          onChange={handleOnChange}
        />
        {isChecked ? labelOn : labelOff}
      </label>
  );
}

export default CheckboxWithLabel;
