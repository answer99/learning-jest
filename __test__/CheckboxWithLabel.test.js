import React from 'react';
import { shallow } from 'enzyme';
import CheckboxWithLabel from '../src/CheckboxWithLabel';


describe('Test Checkbox', () => {
  describe('Rendering', () => {
    test('snapshot test', () => {
      const checkbox = shallow(<CheckboxWithLabel labOn="On" labelOff="Off" />);
      expect(checkbox).toMatchSnapshot('default off checkoutbox');
      
      checkbox.find('input').simulate('change');
      expect(checkbox).toMatchSnapshot('checkoutbox On');

    });
  });

  describe('functional test', () => {
    test('CheckboxWithLabel change the text after click', () => {
      const checkbox = shallow(<CheckboxWithLabel labelOn="On" labelOff="Off" />);

      expect(checkbox.text()).toEqual('Off');

      checkbox.find('input').simulate('change');
    
      expect(checkbox.text()).toEqual('On');

    });
  });
});
