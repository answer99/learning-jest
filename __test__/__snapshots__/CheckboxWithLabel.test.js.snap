// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Checkbox Rendering snapshot test: checkoutbox On 1`] = `ShallowWrapper {}`;

exports[`Test Checkbox Rendering snapshot test: default off checkoutbox 1`] = `ShallowWrapper {}`;
